.eslintrc.js

```diff
@@ -0,0 +1,121 @@
+
+    '@typescript-eslint/no-require-imports': [ 'error' ],
+
+    // 2 space indentation
+    '@typescript-eslint/indent': [ 'error', 2 ],
+
+    // Style
+    'quotes': [ 'error', 'single', { avoidEscape: true } ],
+
+    // ensures clean diffs, see https://medium.com/@nikgraf/why-you-should-enforce-dangling-commas-for-multiline-statements-d034c98e36f8
+    'comma-dangle': [ 'error', 'always-multiline' ],
+
+    // Require all imported dependencies are actually declared in package.json
+    'import/no-extraneous-dependencies': [
+      'error',
+      {
+        optionalDependencies: false,    // Disallow importing optional dependencies (those shouldn't be in use in the project)
+        peerDependencies: false,         // Disallow importing peer dependencies (that aren't also direct dependencies)
+      },
+    ],
+
+    // Require all imported libraries actually resolve (!!required for import/no-extraneous-dependencies to work!!)
+    'import/no-unresolved': [ 'error' ],
+
+    // Require an ordering on all imports
+    'import/order': ['warn', {
+      groups: ['builtin', 'external'],
+      alphabetize: { order: 'asc', caseInsensitive: true },
+    }],
+
+    // Cannot import from the same module twice
+    'no-duplicate-imports': ['error'],
+
+    // Cannot shadow names
+    'no-shadow': ['error'],
+
+    // Required spacing in property declarations (copied from TSLint, defaults are good)
+    'key-spacing': ['error'],
+
+    // Require semicolons
+    'semi': ['error', 'always'],
+
+    // Don't unnecessarily quote properties
+    'quote-props': ['error', 'consistent-as-needed'],
+
+    // No multiple empty lines
+    'no-multiple-empty-lines': ['error'],
+
+    // Max line lengths
+    'max-len': ['error', {
+      code: 120,
+      ignoreUrls: true, // Most common reason to disable it
+      ignoreStrings: true, // These are not fantastic but necessary for error messages
+      ignoreTemplateLiterals: true,
+      ignoreComments: true,
+      ignoreRegExpLiterals: true,
+    }],
+
+    // // One of the easiest mistakes to make
+    // '@typescript-eslint/no-floating-promises': ['error'],
+
+    // Don't leave log statements littering the premises!
+    'no-console': ['error'],
+
+    // Useless diff results
+    'no-trailing-spaces': ['error'],
+
+    // Must use foo.bar instead of foo['bar'] if possible
+    'dot-notation': ['error'],
+
+    // Are you sure | is not a typo for || ?
+    'no-bitwise': ['error'],
+
+    // Member ordering
+    '@typescript-eslint/member-ordering': ['error', {
+      default: [
+        'public-static-field',
+        'public-static-method',
+        'protected-static-field',
+        'protected-static-method',
+        'private-static-field',
+        'private-static-method',
+
+        'field',
+
+        // Constructors
+        'constructor', // = ["public-constructor", "protected-constructor", "private-constructor"]
+
+        // Methods
+        'method',
```

Comment:
I personally prefer having all my fields at the top, followed by the constructor, followed by methods---something like this:
```suggestion
        // Fields
        'field', // = ['public-static-field', 'protected-static-field', 'private-static-field', 'public-instance-field', 'protected-instance-field', 'private-instance-field',
        //             'public-abstract-field', 'protected-abstract-field', private-abstract-field']
        
        // Constructors
        'constructor', // = ['public-constructor', 'protected-constructor', 'private-constructor']
        
        // Methods
        'method' // = ['public-static-method', 'protected-static-method', 'private-static-method', 'public-instance-method', 'protected-instance-method', 'private-instance-method',
        //             'public-abstract-method', 'protected-abstract-method', 'private-abstract-method']
```
But it's up to you, what do you think?

Reply:
i feel like static methods should go before the constructor because they invoke the constructor... I kind of like to Russian doll the code architecturally.. but honestly dont even know if we will use static methods HA

Reply:
Oh my bad I didn't read the static part LMAO okay forget this